src/bin/ion/commands/generate/model.rs

```diff
@@ -221,6 +229,26 @@ impl AbstractDataType {
     }
 }
 
+/// Helper function for serializing abstract data type's `source` field that represents an ISL type.
+/// This method returns the name for the given ISL type.
+// TODO: `IslType` does not implement `Serialize`, once that is available this method can be removed.
+fn serialize_type_name<S>(isl_type: &IslType, serializer: S) -> Result<S::Ok, S::Error>
+where
+    S: Serializer,
+{
+    isl_type
+        .name()
+        .as_ref()
+        .ok_or(S::Error::custom("Isl type doesn't have a name"))?
+        .serialize(serializer)
+}
+
+/// Helper function for checking to skip or serialize `source` field in abstract data type that represents an ISL type.
+/// This method returns true if the ISl type doesn't have a name, otherwise returns false.
+fn has_no_name(isl_type: &IslType) -> bool {
```

Comment:
Consider naming this as a positive assertion (`is_anonymous`) rather than a negative one (`has_no_name`)